cli/trampolines/trampolines_powerpc64le.S

```diff
@@ -0,0 +1,25 @@
+#include "../../src/jl_exported_funcs.inc"
+
+// Notes:
+// bctr: branch to CTR without LR update (tail-call)
+// localentry: On PPC functions have a localentry that assumes r2 contains
+//             the TOC pointer, and a global entry point that sets r2.
+// See 64-Bit ELF V2 ABI Specification: Power Architecture v1.4
+
+#define XX(name) \
+.global name; \
+.type name##, @function; \
+.cfi_startproc; \
+name##: ; \
+    addis 2, 12, .TOC.-name##@ha; \
```

Comment:
I guess the question here is which side of the TOC transition the stubs are on. I take it this means that they are on the callee side? In that case, I agree that this is the way to do it. Also explains why you need to drop the `std 2,24(1)`, since that's supposed to be done by the caller for cross TOC calls. I had assumed that this works like the PLT, which is in the calling module, but I guess it's more like a reverse-PLT.

Reply:
Makes sense!